### PR TITLE
feat(THA-1187): add cumulative protocol fee metric

### PR DESCRIPTION
**Pull request by LawsonGraham**
added a counter for both the cdp and thalaswap projects, where we aggregate net fees for protocol into `cumulative_protocol_fee`

### CODE REVIEW

**Review by 0xbe1 — APPROVED** [review]
lgtm

**Comment by 0xbe1** [comment]
@LawsonGraham can you also attach a link to the data source?

File: projects/mod/src/processor.ts [hunk 1]
```diff
@@ -81,10 +91,15 @@ vault
...
     ctx.eventLogger.emit("update_vault", getVaultUpdatedAttrs(event));
   })
-  .onEventLiquidationEvent((event, ctx) => {
+  .onEventLiquidationEvent(async (event, ctx) => {
     const coin = event.type_arguments[0];
     const coinInfo = getCoinInfo(coin);
 
+    const timestamp = Number(ctx.transaction.timestamp);
+    const priceIn = await getPrice(coinInfo.token_type.type, timestamp);
+    const feeUsd = event.data_decoded.liquidation_fee.scaleDown(coinInfo.decimals).multipliedBy(priceIn);
```
Comment (0xbe1) [inline projects/mod/src/processor.ts:100]: let's not do USD, use coin unit instead

File: projects/thalaswap/src/processors/base_pool.ts [hunk 2]
```diff
@@ -71,6 +71,9 @@ export async function onEventSwapEvent(
 
   volumeGauge.record(ctx, volumeUsd, { poolType, pairTag });
   feeGauge.record(ctx, feeUsd, { poolType, pairTag });
+
+  // we create this fee counter to avoid the need to aggeregate feeGauge to get cumulative fee
+  ctx.meter.Counter("cumulative_protocol_fee").add(feeUsd);
```
Comment (0xbe1) [inline projects/thalaswap/src/processors/base_pool.ts:76]: let's not do USD, use coin unit instead

**Review by 0xbe1 — CHANGES_REQUESTED** [review]

File: projects/mod/src/processor.ts (hunk 1 above)
Comment (LawsonGraham) [reply projects/mod/src/processor.ts:100]: in different places we are fees paid in different coins, and thus pricing / value would be different, so we couldn't simply add to one general counter unless we have some base unit like USD to count in... wdyt best way forward is?

Comment (0xbe1) [reply projects/mod/src/processor.ts]: imagine 2 fee events:
- 10 days ago, 1 APT fee, when 1 APT = $10
- today, 1 APT fee, when 1 APT = $8

we cannot simply add 10 + 8 = 18. instead, we should simply count 2 APT as fee

Comment (LawsonGraham) [reply projects/mod/src/processor.ts]: ah this makes total sense - so then we can record by coin, and then aggregate / sum up in USD on dapp size - how does this sound?

Comment (0xbe1) [reply projects/mod/src/processor.ts]: sgtm

**Review by 0xbe1 — APPROVED** [review]
lgtm!

**Pull request merged**